src/components/media-views.js

```diff
@@ -646,6 +670,17 @@ AFRAME.registerComponent("media-video", {
       this.changeVolumeBy(volumeMod);
     }
 
+    const heldLeftHand = interaction.state.leftHand.held === this.el;
+    const heldRightHand = interaction.state.rightHand.held === this.el;
+    const heldRightRemote = interaction.state.rightRemote.held === this.el;
+    if (
+      (heldLeftHand && userinput.get(interaction.options.leftHand.grabPath)) ||
+      (heldRightHand && userinput.get(interaction.options.rightHand.grabPath)) ||
+      (heldRightRemote && userinput.get(interaction.options.rightRemote.grabPath))
+    ) {
+      this.localSnapCount = 0;
+    }
+
```

Comment:
I recognize this blob because I wrote a similar one. I started writing this kind of check in a nicer way, using a helper caled `isHeld` on the interaction system. What we care about here is whether the object went from not held to held, so now I'd write this as:

```
    const isHeld = interaction.isHeld(this.el);
    if (isHeld && !this.wasHeld) {
      this.localSnapCount = 0;
    }
    this.wasHeld = isHeld;
```

(You will have to pull in the code that has that helper into your branch, which I'm deploying this afternoon in #1203 . See https://github.com/mozilla/hubs/pull/1203/files#diff-519394e566e5f85b23992db2e1b8f609R26 )